runtime/parser/type.go

```diff
@@ -467,6 +478,35 @@ func defineIntersectionOrDictionaryType() {
 			}
 		},
 	)
+
+	// While restricted types have been removed from Cadence, during the first few months of the
+	// migration period, leave a special error in place to help developers
+
+	setTypeMetaLeftDenotation(
+		lexer.TokenBraceOpen,
+		func(p *parser, rightBindingPower int, left ast.Type) (result ast.Type, err error, done bool) {
+
+			// Perform a lookahead
+
+			current := p.current
+			cursor := p.tokens.Cursor()
+
+			// Skip the `{` token.
+			p.next()
+
+			// In case there is a space, the type is *not* considered a restricted type.
+			// The buffered tokens are replayed to allow them to be re-parsed.
+			if p.current.Is(lexer.TokenSpace) {
+				p.current = current
+				p.tokens.Revert(cursor)
+
+				return left, nil, true
+			}
+
+			// It was determined that a restricted type is parsed, so error
+			return nil, p.syntaxError("restricted types have been removed; replace with an intersection type"), true
```

Comment:
Depending on the program, there are two possible resolutions:
- Restricted type is `AnyStruct`/`AnyResource` (e.g. `AnyStruct{I1, ..., In}`): replace with equivalent intersection type (`{I1, ..., In}`)
- Restricted type is concrete type (e.g. `T{I1, ..., In}`): remove restrictions / leave concrete type

```suggestion
			return nil, p.syntaxError("restricted types have been removed; consider using an intersection type or removing the restrictions"), true
```